components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -28,10 +29,50 @@ export class TeamDBImpl implements TeamDB {
         return (await this.getEntityManager()).getRepository<DBTeamMembership>(DBTeamMembership);
     }
 
+    public async findTeamById(teamId: string): Promise<Team | undefined> {
+        const teamRepo = await this.getTeamRepo();
+        return teamRepo.findOne({ id: teamId });
+    }
+
+    public async findMembershipsByTeam(teamId: string): Promise<DBTeamMembership[]> {
+        const membershipRepo = await this.getMembershipRepo();
+        return membershipRepo.find({ teamId });
+    }
+
     public async findTeamsByUser(userId: string): Promise<Team[]> {
         const teamRepo = await this.getTeamRepo();
         const membershipRepo = await this.getMembershipRepo();
         const memberships = await membershipRepo.find({ userId });
         return teamRepo.findByIds(memberships.map(m => m.teamId));
     }
+
+    public async createTeam(userId: string, name: string): Promise<Team> {
+        if (!name) {
+            throw new Error('Team name cannot be empty');
+        }
+        if (!/^[A-Za-z0-9 _-]+$/.test(name)) {
+            throw new Error('Team name should match /^[A-Za-z0-9 _-]+$/');
```

Comment:
**suggestion**: Could we use a friendlier error message when using special characters? Something like the following?

```suggestion
            throw new Error('Please choose a team name with no special characters.);
```

Reply:
Good point! How about:

> Please choose a team name containing only letters, numbers, -, _, or spaces.

Reply:
If I only I had this kind of explanation for every regex expression I ran into. 😅 

Sounds good, as long as we as remove that regex.